### PR TITLE
Don't attempt to json encode string types

### LINKED ISSUE #1394: Question: how to return text/plain response
Is there a better way to return a plain text response vs a json string?

- **I'm submitting a ...**

  - [ ] bug report
  - [ ] feature request
  - [x] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

```
@Get("test")
public async sendText(@Res() ok: TsoaResponse<200, string>): Promise<string> {
  return ok(200, "hello world")
}
```

I would like to send a response like

```
Content-Type: text/plain
hello world
```

## Current Behavior

```
@Get("test")
public async sendText(@Res() response: TsoaResponse<200, string>): Promise<string> {
  return response(200, "hello world")
}
```

and

```
@Get("test")
public async sendText(): Promise<string> {
 return "hello world"
}
```

Both send:

```
Content-Type: application/json; charset=utf-8
"hello world"
```

I also tried:

```
@Get("test")
public async sendText(@Res() response: TsoaResponse<200, string>): Promise<string> {
  return response(200, "hello world", { "Content-Type": "text/plain" })
}
```

which sends

```
Content-Type: text/plain; charset=utf-8
"hello world" (Note the quotes)
```

## Workaround

I've worked around this using;

```
@Get("test")
@Produces("text/plain")
public async sendText(@Request() req: express.Request): Promise<string> {
  req.res?.header("Content-Type", "text/plain; charset=utf-8")
  req.res?.send("hello world")
  return ""
}
```

It feels a bit hacky, I don't like returning the empty string. 

Is there a way of specifying that the TsoaResponse content type is something other than JSON?

Thanks!

### PR DESCRIPTION
**Pull request by keslerm**
### All Submissions:

- [X] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [ ] Have you written unit tests?
- [ ] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [X] This PR is associated with an existing issue?

**Closing issues**
close #1394

### CODE REVIEW

**Comment by keslerm** [comment]
I'm only really familiar with Express but reviewing the code for the other two types of servers they don't appear to have the same issue Express is having here.

**Comment by WoH** [comment]
Hi there, can you please add integration tests to show we have a consistent behavior?

**Comment by WoH** [comment]
Sure, you can add a new controller method here:
https://github.com/lukeautry/tsoa/blob/master/tests/fixtures/controllers/getController.ts#L313

that returns a string.

Then, call them:
https://github.com/lukeautry/tsoa/blob/master/tests/integration/hapi-server.spec.ts
https://github.com/lukeautry/tsoa/blob/master/tests/integration/koa-server.spec.ts
https://github.com/lukeautry/tsoa/blob/master/tests/integration/express-server.spec.ts


Then, verify the response headers.

**Comment by hpx7** [comment]
We are running into this issue as well. Is this fine to merge?

**Comment by keslerm** [comment]
@WoH I got side tracked by work, is there anything I need to do on this still?

**Comment by WoH** [comment]
Can you check the content type headers in the tests? The body could've been correct before, now we want text/html not application/json

**Comment by keslerm** [comment]
@WoH I added some appropriate calls for each server type (hapi and express required an extra step, koa seemed to do it correctly from the git go)

There are a couple of tests failing as a result of this change however. I looked at them a bit and I'm not entirely sure what is ultimately causing them to fail

```
tsoa-tests:   3 failing
tsoa-tests:   1) Express Server
tsoa-tests:        Sub resource
tsoa-tests:          parses path parameters from the controller description:
tsoa-tests:      Uncaught AssertionError: expected {} to equal 'main-123'
tsoa-tests:       at /home/rridley/projects/aegiszero/tsoa/tests/integration/express-server.spec.ts:1450:29
tsoa-tests:       at Test.<anonymous> (integration/express-server.spec.ts:1565:11)
tsoa-tests:       at Test.assert (/home/rridley/projects/aegiszero/tsoa/node_modules/supertest/lib/test.js:172:8)
tsoa-tests:       at Server.localAssert (/home/rridley/projects/aegiszero/tsoa/node_modules/supertest/lib/test.js:120:14)
tsoa-tests:       at Object.onceWrapper (node:events:628:28)
tsoa-tests:       at Server.emit (node:events:514:28)
tsoa-tests:       at Server.emit (node:domain:489:12)
tsoa-tests:       at emitCloseNT (node:net:2148:8)
tsoa-tests:       at processTicksAndRejections (node:internal/process/task_queues:81:21)
tsoa-tests: 
tsoa-tests:   2) Express Server
tsoa-tests:        Sub resource
tsoa-tests:          parses path parameters from the controller description and method description:
tsoa-tests:      Uncaught AssertionError: expected {} to equal 'main-123:sub-456'
tsoa-tests:       at /home/rridley/projects/aegiszero/tsoa/tests/integration/express-server.spec.ts:1459:29
tsoa-tests:       at Test.<anonymous> (integration/express-server.spec.ts:1565:11)
tsoa-tests:       at Test.assert (/home/rridley/projects/aegiszero/tsoa/node_modules/supertest/lib/test.js:172:8)
tsoa-tests:       at Server.localAssert (/home/rridley/projects/aegiszero/tsoa/node_modules/supertest/lib/test.js:120:14)
tsoa-tests:       at Object.onceWrapper (node:events:628:28)
tsoa-tests:       at Server.emit (node:events:514:28)
tsoa-tests:       at Server.emit (node:domain:489:12)
tsoa-tests:       at emitCloseNT (node:net:2148:8)
tsoa-tests:       at processTicksAndRejections (node:internal/process/task_queues:81:21)
tsoa-tests: 
tsoa-tests:   3) Inversify Express Server Dynamic Container
tsoa-tests:        can handle get request with no path argument:
tsoa-tests:      Uncaught AssertionError: expected {} to equal '/v1/ManagedTest'
tsoa-tests:       at /home/rridley/projects/aegiszero/tsoa/tests/integration/inversify-dynamic-container.spec.ts:13:24
tsoa-tests:       at Test.<anonymous> (integration/inversify-dynamic-container.spec.ts:41:11)
tsoa-tests:       at Test.assert (/home/rridley/projects/aegiszero/tsoa/node_modules/supertest/lib/test.js:172:8)
tsoa-tests:       at Server.localAssert (/home/rridley/projects/aegiszero/tsoa/node_modules/supertest/lib/test.js:120:14)
tsoa-tests:       at Object.onceWrapper (node:events:628:28)
tsoa-tests:       at Server.emit (node:events:514:28)
tsoa-tests:       at Server.emit (node:domain:489:12)
tsoa-tests:       at emitCloseNT (node:net:2148:8)
tsoa-tests:       at processTicksAndRejections (node:internal/process/task_queues:81:21)
```

**Comment by WoH** [comment]
I'll take a look, first guess is the data is now on res.text, and it's checking res.body in the tests

File: packages/cli/src/routeGeneration/templates/express.hbs [hunk 1]
```diff
@@ -208,6 +208,9 @@ export function RegisterRoutes(app: Router) {
         if (data && typeof data.pipe === 'function' && data.readable && typeof data._read === 'function') {
             response.status(statusCode || 200)
             data.pipe(response);
+        } else if (data && typeof data === 'string') {
```
Comment (WoH) [inline packages/cli/src/routeGeneration/templates/express.hbs:211]: data could be `''` which is nullish

**Comment by david-loe** [comment]
Any plans on reopening this? `''` could create a `204` in my opinion

**Comment by xseman** [comment]
@WoH Is this still being worked on? I'd love to help if it's still planned.

**Pull request closed without merging**